src/datadog/grok/parse_grok_rules.rs

```diff
@@ -176,12 +176,13 @@ fn parse_pattern(
 ) -> Result<GrokRule, Error> {
     parse_grok_rule(pattern, context)?;
     let mut pattern = String::new();
+    // In Oniguruma the (?m) modifier is used to enable the DOTALL mode(dot includes newlines),
+    // as opposed to the (?s) modifier in other regex flavors.
     // \A, \z - parses from the beginning to the end of string, not line(until \n)
-    pattern.push_str(r"\A");
+    pattern.push_str(r"(?m)\A"); // (?m) enables the DOTALL mode by default
     pattern.push_str(&context.regex);
     pattern.push_str(r"\z");
 
-    // our regex engine(onig) uses (?m) mode modifier instead of (?s) to make the dot match all characters
     pattern = pattern.replace("(?s)", "(?m)").replace("(?-s)", "(?-m)");
```

Comment:
nit: This could be done all in one go:
```rust
    let pattern = [
        // In Oniguruma the (?m) modifier is used to enable the DOTALL mode(dot includes newlines),
        // as opposed to the (?s) modifier in other regex flavors.
        // \A, \z - parses from the beginning to the end of string, not line(until \n)
        r"(?m)\A", // (?m) enables the DOTALL mode by default
        &context.regex.replace("(?s)", "(?m)").replace("(?-s)", "(?-m)"),
        r"\z",
    ].concat();
```

Reply:
Good idea, updated.